File: samples/micronaut-java/src/main/java/io/openapiprocessor/samples/EchoController.java
/*
 * Copyright 2021 https://github.com/openapi-processor/openapi-processor-samples
 * PDX-License-Identifier: Apache-2.0
 */

package io.openapiprocessor.samples;

import io.micronaut.http.annotation.Controller;
import io.openapiprocessor.openapi.api.EchoApi;

@Controller
public class EchoController implements EchoApi {

    @Override
    public String getEcho (String source) {
        return source;
    }
}

File: samples/spring-mvc-boot-4-exchange-kt/src/main/kotlin/io/openapiprocessor/samples/echo/server/EchoController.kt
/*
 * Copyright 2026 https://github.com/openapi-processor/openapi-processor-samples
 * PDX-License-Identifier: Apache-2.0
 */

package io.openapiprocessor.samples.echo.server

import io.openapiprocessor.openapi.server.api.EchoApi
import org.springframework.web.bind.annotation.RestController

@RestController
class EchoController: EchoApi {
    override fun postEcho(reverse: Boolean, body: Echo): Echo {
        return if (reverse) {
            body.reverse()
        } else {
            body
        }
    }
}
